### DISCUSSION TITLE
getting curlCode: 35, SSL connect error [Op:IteratorGetNext] while reading TFRecords dataset from minIO S3 bucket using tf.data.TFRecordDataset(training_files)

### DISCUSSION THREAD

**Discussion by vishal-ranjan** [minio/minio#15073] [Q&A] [answered]
## NOTE
I am getting this error:
tensorflow.python.framework.errors_impl.UnknownError: : curlCode: 35, SSL connect error [Op:IteratorGetNext] while reading TFRecords dataset from minIO S3 bucket using tf.data.TFRecordDataset(training_files)

Here is the complete error message:
 
> Traceback (most recent call last):
File "/home/vishal/minio_tf2.py", line 281, in <module>
batch =next(iter(mapped_ds))
File "/home/vishal/anaconda3/lib/python3.9/site-packages/tensorflow/python/data/ops/iterator_ops.py", line 766, in __next__
return self._next_internal()
File "/home/vishal/anaconda3/lib/python3.9/site-packages/tensorflow/python/data/ops/iterator_ops.py", line 749, in _next_internal
ret = gen_dataset_ops.iterator_get_next(
File "/home/vishal/anaconda3/lib/python3.9/site-packages/tensorflow/python/ops/gen_dataset_ops.py", line 3017, in iterator_get_next
_ops.raise_from_not_ok_status(e, name)
File "/home/vishal/anaconda3/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 7164, in raise_from_not_ok_status
raise core._status_to_exception(e) from None # pylint: disable=protected-access
tensorflow.python.framework.errors_impl.UnknownError: : curlCode: 35, SSL connect error [Op:IteratorGetNext]

## Expected Behavior
I m using model.fit() after loading data from minIO and the training the data. when calling the model.fit(), this error is thrown.

## Steps to Reproduce (for bugs)
I am using code on this link: https://blog.min.io/hyper-scale-machine-learning-with-minio-and-tensorflow/
and when i am calling model.fit() this error is thrown.

## Your Environment

* Version used (`minio --version`): 7.6
* Tensorflow version: 2.9.1

**Comment by harshavardhana** [comment]
> File "/home/vishal/anaconda3/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 7164, in raise_from_not_ok_status
> raise core._status_to_exception(e) from None # pylint: disable=protected-access
> tensorflow.python.framework.errors_impl.UnknownError: : curlCode: 35, SSL connect error [Op:IteratorGetNext]

Your server is running in TLS/SSL but the example code we have tries to talk over HTTP - so you change the code accordingly to your local MinIO deployment.

**Comment by VishalPrasadIITGn** [comment]
What changes to make in the code for that? 
Sorry if the answer is trivial, I am a bit new to this.

**Comment by klauspost** [comment, marked as answer]
```
minioClient = Minio(minio_address,
                  access_key=minio_access_key,
                  secret_key=minio_secret_key,
                  secure=True)
```                  
.. would be my guess.

**Reply by vishal-ranjan** [reply]
Hi. I am already using this in my code. Anythin else to add in my code that might be helpful?
